matchtree.go

```diff
@@ -292,12 +293,19 @@ func (t *symbolSubstrMatchTree) prepare(doc uint32) {
 			continue
 		}
 
-		if end <= sections[secIdx].End {
-			t.current[0].symbol = true
-			t.current[0].symbolIdx = uint32(secIdx)
-			trimmed = append(trimmed, t.current[0])
+		if end > sections[secIdx].End {
+			t.current = t.current[1:]
+			continue
+		}
+
+		if t.anchored && !(start == sections[secIdx].Start && end == sections[secIdx].End) {
+			t.current = t.current[1:]
+			continue
 		}
 
+		t.current[0].symbol = true
+		t.current[0].symbolIdx = uint32(secIdx)
+		trimmed = append(trimmed, t.current[0])
```

Comment:
nice, I think this code is clearer where you all the conditionals are reasons to exclude the current match.